lib/drivers/onetouch/oneTouchVerio.js

```diff
@@ -0,0 +1,730 @@
+      (err) => {
+        if (err) {
+          debug(`retrieveData: Error writing to device '${this.devicePath}':`, err);
+          return;
+        }
+        // debug(`retrieveData: wrote ${numWritten} bytes:`, requestData);
+        const readBuffer = Buffer.alloc(BLOCKDEVICE_BLOCKSIZE);
+        fs.read(
+          this.fileHandle, readBuffer, 0, BLOCKDEVICE_BLOCKSIZE, seekOffset,
+          (err2) => {
+            if (err2) {
+              debug(`retrieveData: Error reading from device '${this.devicePath}':`, err2);
+              return;
+            }
+            // debug(`retrieveData: read ${numRead} bytes:`, readBuffer);
+            callback(readBuffer);
+          },
+        );
+      },
+    );
+  }
+}
+
+
+class OneTouchVerio {
+  constructor(cfg) {
+    this.cfg = cfg;
+    if (process.platform === 'linux') {
+      this.communication = new USBScsiDevice();
+    } else {
+      this.communication = new BlockDevice();
+    }
+  }
+
+  openDevice(deviceInfo, callback) {
+    return this.communication.openDevice(deviceInfo, callback);
+  }
+
+  closeDevice(callback) {
+    return this.communication.closeDevice(callback);
+  }
+
+  checkDevice(callback) {
+    return this.communication.checkDevice(callback);
+  }
+
+  static parseApplicationLayer(data, callback) {
+    // const serviceId = data[0];
+    const responseCode = data[1];
+    if (responseCode !== RESPONSE_OK) {
+      debug('parseApplicationLayer: Invalid response code:', responseCode);
+      return;
+    }
+    const commandData = data.slice(2);
+    if (callback) {
+      callback(commandData);
+    }
+  }
+
+  static parseLinkLayer(data, callback) {
+    if (data.length !== USB_BULK_BLOCKSIZE) {
+      debug('parseLinkLayer: Invalid data blocksize:', data.length);
+      return;
+    }
+    if (data[0] !== STX) {
+      debug('parseLinkLayer: Invalid start byte:', data[0]);
+      return;
+    }
+    const length = data.readUInt16LE(1);
+    if (data[length - 3] !== ETX) {
+      debug('parseLinkLayer: Invalid end byte:', data[length - 3]);
+      return;
+    }
+    const crc16 = data.readUInt16LE(length - 2);
+    const calculatedCrc16 = crc.calcCRC_A(data.slice(0, length - 2), length - 2);
+    if (crc16 !== calculatedCrc16) {
+      debug('parseLinkLayer: CRC error:', crc16, '!=', calculatedCrc16);
+      return;
+    }
+    const applicationData = data.slice(3, length - 3);
+    callback(applicationData);
+  }
+
+  retrieveData(lbaNumber, linkLayerRequestData, callback) {
+    this.communication.retrieveData(lbaNumber, linkLayerRequestData, (linkResponseData) => {
+      this.constructor.parseLinkLayer(linkResponseData, (applicationResponseData) => {
+        this.constructor.parseApplicationLayer(applicationResponseData, (commandData) => {
+          callback(commandData);
+        });
+      });
+    });
+  }
+
+  static buildLinkLayerFrame(applicationData) {
+    const length = 3 + applicationData.length + 3;
```

Comment:
Please add comments as to what the `3`'s mean, or add a const somewhere (eg `const MESSAGE_ENVELOPE = 3;`) that makes the meaning explicit.

Reply:
I'll leave it to @DorianScholz to define these constants, as he probably has a better idea what they should be :)

Reply:
Ping @DorianScholz